implants/lib/pb/build.rs

```diff
@@ -2,7 +2,71 @@ use std::env;
 use std::path::PathBuf;
 use which::which;
 
+fn get_pub_key() {
+    // Check if IMIX_SERVER_PUBKEY is already set
+    if std::env::var("IMIX_SERVER_PUBKEY").is_ok() {
+        println!("cargo:warning=IMIX_SERVER_PUBKEY already set, skipping fetch");
+        return;
+    }
+
+    // Get the callback URI from environment variable, default to http://127.0.0.1:8000
+    let callback_uri =
+        std::env::var("IMIX_CALLBACK_URI").unwrap_or_else(|_| "http://127.0.0.1:8000".to_string());
+
+    // Construct the status endpoint URL
+    let status_url = format!("{}/status", callback_uri);
+
+    // Make a GET request to /status
+    let response = match reqwest::blocking::get(&status_url) {
+        Ok(resp) => resp,
+        Err(e) => {
+            println!("cargo:warning=Failed to connect to {}: {}", status_url, e);
+            return;
+        }
+    };
+
+    if !response.status().is_success() {
+        println!(
+            "cargo:warning=Failed to fetch status from {}: HTTP {}",
+            status_url,
+            response.status()
+        );
+        return;
+    }
+
+    let json = match response.json::<serde_json::Value>() {
+        Ok(json) => json,
+        Err(e) => {
+            println!(
+                "cargo:warning=Failed to parse JSON response from {}: {}",
+                status_url, e
+            );
+            return;
+        }
+    };
+
+    let pubkey = match json.get("Pubkey").and_then(|v| v.as_str()) {
+        Some(key) => key,
+        None => {
+            println!(
+                "cargo:warning=Pubkey field not found in response from {}",
+                status_url
+            );
+            return;
+        }
+    };
+
+    // Set the IMIX_SERVER_PUBKEY environment variable for the build
+    println!("cargo:rustc-env=IMIX_SERVER_PUBKEY={}", pubkey);
```

Comment:
How does it deal with formatting? I think it's preferrable to use an API like `std::env::var` to ensure this gets set to an env var instead of string formatting into something that looks like a CLI arg, otherwise feels like this could be a vector for command injection